modules/javamail/src/main/scala/emil/javamail/internal/ConnectionResource.scala

```diff
@@ -27,8 +27,9 @@ object ConnectionResource {
 
   def make[F[_]: Sync](mc: MailConfig, settings: Settings): F[JavaMailConnection] =
     Sync[F].delay {
-      val session = createSession(mc, settings)
-      if (mc.urlParts.protocol.toLowerCase.startsWith("imap")) {
+      val session  = createSession(mc, settings)
+      val protocol = mc.urlParts.protocol.toLowerCase
+      if (protocol.startsWith("imap") || protocol.startsWith("gimap")) {
```

Comment:
Ah, so in order to use this, you need to select java.mails gimap implementation by using the `gimap` protocol? Maybe we can add some lines to the docs about these things (I mean later for the master branch)

Reply:
Yeah, we should document this eventually. But first we would need to verify the functionality and the interfaces in practice before we cast them in stone by exposing them publicly and documenting them :wink:

Reply:
Totally agree; I really like it this way.